### ISSUE TITLE
pip install fails on python 3.6

### ISSUE THREAD

**Issue by ErnstHaagsman** [aio-libs/yarl#27]
I'm currently trying to build a python script which uses yarl. When using a python 3.5 virtualenv there is no problem, and `pip install yarl` installs version 0.7.0 without a problem. In a python 3.6 virtualenv using pip 9.0.1, an exception is thrown in yarl's setup.py:

```
Running setup.py (path:/tmp/pip-build-ms_rhff9/yarl/setup.py) egg_info for package yarl
    Running command python setup.py egg_info
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ms_rhff9/yarl/setup.py", line 97, in <module>
        setup(**args)
      File "/usr/lib/python3.6/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/dist.py", line 268, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/dist.py", line 313, in fetch_build_eggs
        replace_conflicting=True,
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 836, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1081, in best_match
        return self.obtain(req, installer)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1093, in obtain
        return installer(requirement)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/dist.py", line 380, in fetch_build_egg
        return cmd.easy_install(req)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 617, in easy_install
        not self.always_copy, self.local_index
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/package_index.py", line 601, in fetch_distribution
        return dist.clone(location=self.download(dist.location, tmpdir))
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/package_index.py", line 517, in download
        found = self._download_url(scheme.group(1), spec, tmpdir)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/package_index.py", line 758, in _download_url
        return self._attempt_download(url, filename)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/package_index.py", line 764, in _attempt_download
        headers = self._download_to(url, filename)
      File "/home/eh/crawler-3.6/lib/python3.6/site-packages/setuptools/package_index.py", line 680, in _download_to
        block = fp.read(bs)
      File "/usr/lib/python3.6/http/client.py", line 449, in read
        n = self.readinto(b)
      File "/usr/lib/python3.6/http/client.py", line 493, in readinto
        n = self.fp.readinto(b)
      File "/usr/lib/python3.6/socket.py", line 586, in readinto
        return self._sock.recv_into(b)
      File "/usr/lib/python3.6/ssl.py", line 1002, in recv_into
        return self.read(nbytes, buffer)
      File "/usr/lib/python3.6/ssl.py", line 865, in read
        return self._sslobj.read(len, buffer)
      File "/usr/lib/python3.6/ssl.py", line 625, in read
        v = self._sslobj.read(len, buffer)
    ConnectionResetError: [Errno 104] Connection reset by peer
```

Retrying it several times didn't change the behavior. After switching back to the 3.5 virtualenv I could uninstall and reinstall yarl there, and then in the 3.6 virtualenv the same error occurred. I'm not sure if it's a pip bug, but other packages seem to install fine. 

Python 3.6.0b2 from Ubuntu PPA
Xubuntu 16.10
pip 9.0.1
yarl 0.6.0 and 0.7.0, haven't tested earlier versions

**Comment by asvetlov** [comment]
I don't know what is exactly going on but since Python 3.6 is not released yet yarl has no Python 3.6 builds.
You might clone the repo and install yarl from local copy by `pip install /path/to/cloned/yarl`.

From other hand `ConnectionResetError` at downloading looks very strange.
I have no idea about the error but pretty sure it's not related to yarl itself.

**Comment by ErnstHaagsman** [comment]
Thank you! That worked
